src/Middleware/RequestThrottling/test/TestUtils.cs

```diff
@@ -16,10 +15,12 @@ public static RequestThrottlingMiddleware CreateTestMiddleware(int? maxConcurren
         {
             var options = new RequestThrottlingOptions
             {
-                MaxConcurrentRequests = maxConcurrentRequests,
+                MaxConcurrentRequests = maxConcurrentRequests == 0 ? 999 : maxConcurrentRequests,
                 RequestQueueLimit = requestQueueLimit
             };
 
+            options.ServerAlwaysBlocks = maxConcurrentRequests == 0;
```

Comment:
This check should be in options or the middleware. Right now this only work if you use TestUtils.

Reply:
I think I understand now. What you have is fine, just use a different method for the ServerAlwaysBlocks case.